File: 03.REST_JAVA_COMUNI_GR02/02. CLIWEB/WS_CONV_UNI_RESTFUL_JAVA_WEB/servicio/conversion_servicio.py
import requests

class ConversionServicio:

    def convertir(self, valor, origen, destino):

        try:
            url = f"http://localhost:8080/WS_CONV_UNI_RESTFUL_JAVA/webresources/conversion/{valor}/{origen}/{destino}"
            response = requests.get(url)

            return response.text

        except:
            return "❌ Error al conectar con el servidor"

File: 03.REST_JAVA_COMUNI_GR02/02. CLIMOV/WS_CONV_UNI_RESTFUL_JAVA_MOVIL/app/src/main/java/ec/edu/gr02/servicio/ConversionServicio.java
package ec.edu.gr02.servicio;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class ConversionServicio {

    public String convertir(double valor, String origen, String destino) {

        try {
            String ruta = "http://10.0.2.2:8080/WS_CONV_UNI_RESTFUL_JAVA/webresources/conversion/"
                    + valor + "/" + origen + "/" + destino;

            URL url = new URL(ruta);
            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
            conn.setRequestMethod("GET");

            BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
            String resultado = in.readLine();

            in.close();

            return resultado;

        } catch (Exception e) {
            return "Error conexión";
        }
    }
}